---
title: A Brief Introduction to Genome Analysis
author: emarro
math: true
---

# What the Heck is a "Genome"?
Whenever I do one of my "Computational Biology for Computer Scienctists" crash courses, we always cover the same crucial (simplified) background information. DNA molecules are stored in your cells, and they have an order to them. This order allows them to form a sequence, and that sequence encodes information that is then used to produce proteins, little biological robots that go about and perform specific actions (kind of like microservices). We can think of a gene as a unit that encodes a single protein in it's entirety, and so the genome is a collection of all the DNA in someone, including  these genes and parts of the DNA surronding those genes. The protiens encoded in your genome underpin basically every single process that is currently undergoing in your body, and so understanding the genome would probably help you a lot in understanding why an organism works the way they work. This was actually the impetus of the human genome project, the idea was that if we are able to sequence and read the whole of the human genome, we could unlock secrets untold about how we humans work and how to treat a number of our illnesses.  


Well as of the date of writing this, genetic diseases still exist and we have to eradicate diseases, so did the human genome project fail to yield useful information or was the genome just a lot harder to work with than initially anticipated it? So if the genome is a collection of DNA sequences that code for proteins, why can't we just compare and analyze strings of letters? How do we analyze it and why is it so difficult?

## Isn't it Just Comparing Strings?

A few of the factors that make genome analysis difficult would be:
1. The sheer size of the human genome (a sequence of ~3 Billion)
   2. The fact that even defining a gene is deceptivly difficult .
   1. If you ask different geneticists, you'll get different answers about the exact number of genes!
   2. How do we deal with variants? If we change a single character in a gene, is that a new gene? 
   3. What happens when the sequence around the gene change the way the gene is expressed? Is that a different gene?
3. Your genome itself isn't the sole thing responsible for the expression of genes,  your genome actually interacts with environmental factors.

Just imagine starting off trying to analyze the first 2 sequenced genomes with no prior information. You would first notice that you have two strings of slightly differnt of lengths $\approx 3\times10^{9} + \epsilon$, since biological factors can lead to insertions or deletions of parts of the sequence. The first thing you might try to do is align them, since both of your samples are from ostensibly living adult humans, their genomes can't be that different right? 
Well, kind of. Humans share $\approx 99.9\%$ of the genome, and so most of the genetic variation in humans comes from that $\approx 0.1\%$ of the differences. But even then, that's still $\approx 2-5$ million differences or so. So we have to string of different lengths that are mostly the same, can we just ignore the differences? No actually, it turns out that those few differences, are actually vitally important as they are one of the main components that leads to variation amongst humans! So we can ignore everything else and keep only the differences? Well sort of, the differences aren't the only things that matter in the genome we also care about _where_ the differences are and what the sequences adjacent to the differences are. So for some specific questions, sure we can only look at the differences, but in general we cannot ignore all the rest of the genome. 

## So what do we do?
Ok so what exactly are the important parts of the genome? Can we focus on those? Well that depends on your definition of important. We used to call parts of the genome that we didn't know the purpose of "Junk DNA". We later stopped using this name as we kept finding "Junk DNA" that did indeed have a purpose, albiet sometimes that purpose was very much not obvious. So we can't ignore anything, we need to work with the whole unwieldy string? Not necessarily, as we keep studying the genome and identifying the functions of different portions of it, we get an understanding of the structure and patterns of the genome. We can get an idea for which genes tend to interact with each, which regions of DNA impact a given gene, and which pieces of DNA are unlikely to have any sort of impact. The more we study the genome, the more structure we find and so the better analysis we can do. This is what we do in genome analysis, we leverage certain structures or properties of the genome we know such as the rate of mutation of certain regions, the patterns of co-occurence of certain genomic features, and the interactions of the regulatory elements of the genome with respect to each other and our genes of interest. 

### GWAS
One of the more popular and important methods for genome analysis is the aptly named Genome Wide Association Study (GWAS). This type of study works exactly as you might expect; we find two groups of people, a group of people showing a phenotype(trait) of interest and a control group. Assuming a suffecient population size, we can then compare both groups and see if there is a specific genomic feature that the people in the test group have different than the people in the control group. 
If we can identify some features like that, we can then say that the features we observed are __correlated__ with the phenotype of interest, these features are "associated" with the phenotype of interest. 


Now granted, we know a lot more about the genome now than we ever have. Ever since the human genome project and the subsequent rise of RNA sequencing and other techniques to assay the genome, we've learned a whole heck of a lot about the structures and patterns present in the genome, and this has provided some awesome opportunites for the enterprising computer scientist. When we do a sequencing study, we generate gigabytes to terabytes of data corresponding to the DNA of our participants. 

We know significantly more now than we used to about the general structure of DNA though, such as 
1. which regions have higher or lower mutation rates (it varies by location and by the base pairs present at that location!) 
2. patterns of co-occurence/linkage disequillibrium, 
3. better understanding of the regulatory environemnt of the human genome. 